src/main/java/com/gluonhq/richtextarea/model/ParagraphDecoration.java

```diff
@@ -183,11 +183,13 @@ public Builder leftInset(double leftInset) {
 
         public Builder indentationLevel(int indentationLevel) {
             this.indentationLevel = indentationLevel;
+            this.graphicType = indentationLevel == 0 ? GraphicType.NONE : graphicType;
             return this;
         }
 
         public Builder graphicType(GraphicType graphicType) {
             this.graphicType = graphicType;
+            this.indentationLevel = Math.max(graphicType != GraphicType.NONE ? 1 : 0, indentationLevel);
```

Comment:
I would reverse the condition to `graphicType == GraphicType.NONE ? 0 : 1`

Reply:
done